files/default/neutron-ha-tool.py

```diff
@@ -964,6 +964,27 @@ def pick(self):
         return self.agents_by_id[agent_id]
 
 
+class SingleAgentPicker(object):
+    agent_keys = ('id', 'host')
+    agent_selection_value = None
+
+    def __init__(self, qclient, agents):
+        self.agents = agents
+
+    def find_agent(self, value):
+        for agent in self.agents:
+            agent_values = [agent.get(key) for key in self.agent_keys]
+            if value in agent_values:
+                return agent
+        raise IndexError('Cannot find desired agent')
```

Comment:
Yes please, otherwise Murphy's Law dictates we'll inevitably see this and end up in a situation which is impossible to debug.